test/tests/test_setup_tests.py

```diff
@@ -287,6 +288,43 @@ def test_for_circular_variable_references(self):
         with self.assertRaises(variables.VariableError):
             resolve_permutations(test, {}, {})
 
+    def test_finalize(self):
+
+        plugins.initialize_plugins(self.pav_cfg)
+
+        cfg = self._quick_test_cfg()
+
+        cfg['run']['cmds'] = [
+            'echo "{{sys.host_name}}"'
+        ]
+
+        cfg['results'] = {
+            'regex': [{
+                'key': 'foo',
+                'regex': '{{sys.host_name}}',
+            }]
+        }
+
+        test = self._quick_test(cfg, 'finalize_test',
+                                build=False, finalize=False)
+
+        test.build()
+
+        undefered_sys_vars = system_variables.SysVarDict(
+            defer=False,
+            unique=True,
+        )
+
+        fin_var_man = variables.VariableSetManager()
+        fin_var_man.add_var_set('sys', undefered_sys_vars)
+
+        test.finalize(fin_var_man)
+
+        results = test.gather_results(test.run())
+        test.save_results(results)
+
+        plugins._reset_plugins()
+
     def test_resolve_all_vars(self):
```

Comment:
Consider updating the test name to match the new function name (resolve_config).